README.md

```diff
@@ -59,12 +63,143 @@ $ npm i @mixmaxhq/promise-pool
 
 Changelog
 ---------
+
+* 2.0.0 Add `maxPending` option to avoid problematic usage (see new [Troubleshooting](#troubleshooting) section)
+
 * 1.1.1 Move `ava` and `ava-spec` to `devDependencies`.
 
 * 1.1.0 Adds transpilation so it can be used in Node 6 (and prior) environments.
 
 * 1.0.0 Initial release.
 
+Troubleshooting
+---------------
+
+### `cannot queue function in pool`
+
+If you're getting this error, then you're calling `start` too many times
+concurrently without waiting for the previous `start` call to resolve. Please
+read on - one of two things is happening:
+
+- you're using promise-pool in a way that will defeat its purpose
+- you're using promise-pool in an uncommon manner that works with its purpose (but might benefit from being refactored)
```

Comment:
What do you think about describing the problems in a more specific and technical way (than "strange", "uncommon", "unorthodox", etc)? Might be easier for readers to grasp as well as help them feel better if they hit the error. Something like:

```markdown
If you're getting this error, then you're calling `start` too many times
concurrently. Please read on - one of two things is happening:

- you're not waiting for the previous `start` call to resolve before calling `start` again
- you're adding items to the pool in multiple places
```

I'd suggest doing similar throughout the changeset.

Reply:
652d12b